ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/statetransition/epoch/EpochProcessorAltair.java

```diff
@@ -146,4 +155,29 @@ public void processInactivityUpdates(
   protected int getProportionalSlashingMultiplier() {
     return specConfigAltair.getProportionalSlashingMultiplierAltair();
   }
+
+  private Optional<SszList<SszByte>> getZeroParticipationFlags(final BeaconStateAltair state) {
+    SszList<SszByte> flags = zeroParticipationFlags;
+    SszMutableList<SszByte> mutableFlags;
+    if (flags == null) {
+      mutableFlags = state.getCurrentEpochParticipation().getSchema().of().createWritableCopy();
+      mutableFlags.setAll(SszByte.ZERO, state.getValidators().size());
+      flags = mutableFlags.commitChanges();
+      zeroParticipationFlags = flags;
```

Comment:
nit: Can move this line to just before the return and avoid the duplication. In fact both the flags and zeroParticipationFlags assignments could move down.

Reply:
id prefer not to, in this case we're only going mutable because we need to actually change the object, moving it outside cut the improvement in half because we're getting mutable then storing constantly...